roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,49 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerized
+
+class DockerImageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+
+        args = {
+            "name": self.rpm_docker_images(),
+        }
+
+        docker_images = self.module_executor("docker_image_facts", args, tmp, task_vars)
+
+        if len(docker_images['images']) == 0:
+            return {"failed": True, "msg": "Unable to find any docker images!"}
+
+        # if we find any docker images, attempt to do 1-to-1 match between
+        # each image description name key and items from rpm_docker_images
+        matched_images = set()
+        for i in docker_images['images']:
```

Comment:
Please don't call it `i`. `i` is a good name for indices in a list... here `image` is probably better.